src/main/java/kalchenko/task/TaskController.java

```diff
@@ -0,0 +1,73 @@
+package kalchenko.task;
+
+import kalchenko.exception.TaskNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class TaskController {
```

Comment:
Пожалуйста, добавь валидацию.

Reply:
У меня не получается сделать валидацию. Все аннотации связанные с валидацией просто игнорируются. Решение подобных проблем, которые нашёл не помогли пофиксить это.  Игнорируется как валидация у методов, так и валидация у полей.

Reply:
Первое, что я бы проверил - это модификаторы доступа функций. Я ниже уже отметил, что лучше сделать методы публичными, чтобы аннотации спринга с ними точно работали. Второе - проверить расстановку аннотаций: у полей класса, используемого как RequestBody, проставить валидационные аннотации (`@Min()`, `@NotNull`, `@NotBlank`, etc), а уже в самих методах использовать аннотацию `@Valid`. Я вижу, что иногда используется аннотация `@Validated`, но в ней нет необходимости на текущий момент.

Reply:
Починил валидацию.
Заглянув в логи проекта, увидел, что spring не может инициализировать классы связанные с валидацией с ошибкой: `Failed to set up a Bean Validation provider javax.validation.NoProviderFoundException: Unable to create a Configuration, because no Jakarta Bean Validation provider could be found. Add a provider like Hibernate Validator (RI) to your classpath.`. Это ошибка означает, что в `pom.xml` не подключены зависимости для Hibernate Validator, но они у меня были подключены. Проблема была в версии Hibernate. Я выбрал самую последнюю версию, которая указана на [сайте Maven](https://mvnrepository.com/artifact/org.hibernate.validator/hibernate-validator), но почему-то  это не работало. Пришлось выбрать версию 6.1.7, с ней уже всё заработало. 
Изменил модификаторы доступа, оставил `@Validated`, так как @`Valid` просто-напросто не работает с параметрами запроса.

Reply:
Тогда я предложил бы использовать `spring-boot-starter-validaton`, вместо валидации Hibernate. Последний - фреймворк для JPA, а работы с бд у нас в текущем задании нет.

Reply:
А насчет аннотаций - судя по документации, `@Valid` как раз должен работать с параметрами, только не запроса, а метода. Но не будет работать, если пометить им класс, который предполагается валидировать. Может в этом проблема. `@Validated` как правило, используют для того, чтобы группировать валидацию - то есть, когда класс используется один, но валидировать его нужно по разному ([вот тут хороший пример показан](https://stackoverflow.com/a/52092984))